Dev

n1kolayvas1lev opened the pull request:
Зачётное задание по курсу PythonPY110
Васильев Николай Георгиевич

n1kolayvas1lev committed:
add .gitignore

n1kolayvas1lev committed:
add requirements.txt

n1kolayvas1lev committed:
14.12.2021 test @ 20.16

n1kolayvas1lev committed:
14.12.2021 test @ 20.34

n1kolayvas1lev committed:
14.12.2021 test @ 21.19

n1kolayvas1lev committed:
14.12.2021 test @ 21.22

n1kolayvas1lev committed:
14.12.2021 test @ 21.24

aeksei commented on main.py:
```diff
@@ -0,0 +1,73 @@
+
+def title():
+    n = random.randint(1,5)
+#    with open('books.txt', 'r', encoding='utf-8') as books:
+#    books.readline(n)
```
```python
import random

with open('books.txt', 'r', encoding='utf-8') as books:
    return random.choice(books.readlines())
```

aeksei reviewed (changes requested):
Исправить генерацию случайных книг

n1kolayvas1lev committed:
14.12.2021 test @ 21.49

n1kolayvas1lev committed:
14.12.2021 test @ 23.04

n1kolayvas1lev committed:
14.12.2021 test @ 23.25

n1kolayvas1lev commented:
У меня невыносимые проблемы с аннотацией типов. Например, стр. 33, при попытке написать нечто вроде:
def year(): -> int:
    return random.randint(900, 2021)
Я получаю ошибку, гласящую, что return находится вне функции. Я пытался донести своим "-> int", что желаю видеть на выходе у функции целое число. При этом контекстные подсказки, отображают аннотацию вполне правильно. Я совершенно не понимаю, что я делаю не так.

aeksei reviewed (changes requested):
- [x] добавить аннотацию типов
- [x] добавить документацию к функциям
- [x] заменить два момента на list comprehension (на них не могу дать указание из-за технических особенностей, а не из вредности)

После исправлений можно сливать ветку и закрывать pull request

---

Если стоит статус **Changes requested**, то нужно в данную ветку внести обязательные изменения, которые представлены ниже. Если выше представлены комментарии, то они носят рекомендательный характер.
После нужно сделать "push", тогда коммиты с изменениями сразу попадут в "pull request".

n1kolayvas1lev committed:
23.12.2021
- Добавлена аннотация типов;
- Добавлена документация к функциям;
- Заменены циклы на List comp.

n1kolayvas1lev commented:
Готово.
